core/services/ping/ping360_ethernet_prober.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+"""
+Scan with ./findPing360.py [current IP of the interface to scan]
+"""
+
+import asyncio
+import socket
+from typing import List
+
+import psutil
+
+
+def list_ips() -> List[str]:
+    """
+    Returns a list of IPs found on ethernet interfaces that are currently up
+    """
+    stats = psutil.net_if_stats()
+
+    available_networks = []
+    ips = []
+    for interface in stats.keys():
+        if interface in stats and getattr(stats[interface], "isup"):
+            available_networks.append(interface)
+
+    for interface in available_networks:
+        phys_list = list(filter(lambda address: address.family == socket.AF_INET, psutil.net_if_addrs()[interface]))
+        ips.extend([phys.address for phys in phys_list])
```

Comment:
- avoid repeated `psutil` calls
- clean up with filtered generator expression
- avoid needless list creation

```suggestion
    network_interface_addresses = psutil.net_if_addrs()
    for interface in available_networks:
        ips.extend(phys.address for phys in network_interface_addresses[interface] if phys.family == socket.AF_INET)
```

Reply:
nice catch